### PR TITLE
plugins/mkdnflow: init + test

### LINKED ISSUE #509: Plugin request: mkdnflow.nvim
**Plugin**: mkdnflow.nvim
**Plugin repository**: https://github.com/jakewvincent/mkdnflow.nvim
**Is the plugin on nixpkgs**: No

## Extra information
The configuration for the plugin can be found [here](https://github.com/jakewvincent/mkdnflow.nvim#%EF%B8%8F-configuration).

### PR DESCRIPTION
**Pull request by GaetanLepage**
Add support for plugin [mkdnflow](https://github.com/jakewvincent/mkdnflow.nvim).

~~We have to wait for the [PR](https://github.com/NixOS/nixpkgs/pull/248356) that adds it upstream [to land in `nixos-unstable`](https://nixpk.gs/pr-tracker.html?pr=248356).~~ -> Done !

Fixes #509

### CODE REVIEW

File: plugins/utils/mkdnflow.nix [hunk 1]
```diff
@@ -0,0 +1,612 @@
...
+          Neovim's working directory.
+
+          - `true`: Changes in perspective will be reflected in the nvim working directory.
+            (In other words, the working directory will "heel" to the plugin's perspective.)
+            This helps ensure (at least) that path completions (if using a completion plugin with
+            support for paths) will be accurate and usable.
+          - `false` (default): Neovim's working directory will not be affected by Mkdnflow.
+        '';
+
+        update = helpers.defaultNullOpts.mkBool true "";
```
Comment (traxys) [inline plugins/utils/mkdnflow.nix:145]: Missing the description ?

```
Determines whether the plugin looks to determine if a followed link is in a different notebook/wiki than before. If it is, the perspective will be updated. Requires root_tell to be defined and priority to be root.

 - true (default): Perspective will be updated when following a link to a file in a separate notebook/wiki (or navigating backwards to a file in another notebook/wiki).
 - false: Perspective will be not updated when following a link to a file in a separate notebook/wiki. Under the hood, links in the file in the separate notebook/wiki will be interpreted relative to the original notebook/wiki.
```

File: plugins/utils/mkdnflow.nix [hunk 2]
```diff
@@ -0,0 +1,612 @@
...
+          `root` but no root directory is found).
+
+          - `first` (default): Links will be interpreted relative to the first-opened file (when
+            the current instance of Neovim was started)
+          - `current`: Links will be interpreted relative to the current file
+          - `root`: Links will be interpreted relative to the root directory of the current
+            notebook (requires `perspective.root_tell` to be specified)
+        '';
+
+        rootTell = helpers.defaultNullOpts.mkNullable (with types; either bool str) "false" ''
```
Comment (traxys) [inline plugins/utils/mkdnflow.nix:127]: maybe either str or (enum [false]) as is done later

**Review by traxys — CHANGES_REQUESTED** [review]

File: plugins/utils/mkdnflow.nix (hunk 1 above)
Comment (GaetanLepage) [reply plugins/utils/mkdnflow.nix]: Indeed !

File: plugins/utils/mkdnflow.nix (hunk 2 above)
Comment (GaetanLepage) [reply plugins/utils/mkdnflow.nix:127]: Done

**Pull request merged**